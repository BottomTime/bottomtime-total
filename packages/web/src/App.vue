<template>
  <div v-if="isLoading"></div>
  <div v-else>
    <SnackBar />
    <NavBar />
    <RouterView />
    <PageFooter />
    <CookieWarning />
  </div>
</template>

<script lang="ts" setup>
import { onBeforeMount, ref } from 'vue';

import CookieWarning from './components/main/CookieWarning.vue';
import { Dispatch, useStore } from '@/store';
import { inject } from '@/helpers';
import NavBar from '@/components/main/NavBar.vue';
import PageFooter from '@/components/main/PageFooter.vue';
import SnackBar from '@/components/main/SnackBar.vue';
import { ApiClientKey } from '@/injection-keys';

const client = inject(ApiClientKey);
const store = useStore();
const isLoading = ref(true);

onBeforeMount(async () => {
  try {
    const currentUser = await client.users.getCurrentUser();
    await store.dispatch(Dispatch.SignInUser, currentUser);
  } catch (error) {
    console.error(error);
  } finally {
    isLoading.value = false;
  }
});
</script>

<style>
@import './assets/bulmaswatch.min.css';
</style>
