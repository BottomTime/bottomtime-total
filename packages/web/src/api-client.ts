import { ApiClient } from '@bottomtime/api';

import { InjectionKey, inject } from 'vue';

export const ApiClientKey: InjectionKey<ApiClient> = Symbol('ApiClient');
export function useClient(): ApiClient {
  const client = inject(ApiClientKey);

  if (!client) {
    throw new Error(
      'Api Client has not been initialized. Did you forget to call `app.provide(key, client)`?',
    );
  }

  return client;
}
