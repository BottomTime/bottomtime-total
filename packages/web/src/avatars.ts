import { AvatarSize, ListAvatarURLsResponseDTO } from '@bottomtime/api';

export type AvatarURLs = ListAvatarURLsResponseDTO['sizes'];

export function getAvatarURL(
  baseUrl: string | undefined,
  size: AvatarSize,
): string | undefined {
  if (!baseUrl) return undefined;

  if (/^\/api\/users\/[a-z0-9]+([_.-][a-z0-9]+)*\/avatar/i.test(baseUrl)) {
    return baseUrl.endsWith('/') ? `${baseUrl}${size}` : `${baseUrl}/${size}`;
  }

  return baseUrl;
}
