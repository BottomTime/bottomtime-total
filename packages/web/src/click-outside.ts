import { Directive, DirectiveBinding } from 'vue';

type HTMLElementWithClickOutside = HTMLElement & {
  clickOutsideEvent: (e: MouseEvent) => void;
};

export const clickOutside: Directive = {
  beforeMount: (el: HTMLElementWithClickOutside, binding: DirectiveBinding) => {
    el.clickOutsideEvent = (event) => {
      // here I check that click was outside the el and his children
      if (!(el == event.target || el.contains(event.target as Node))) {
        // and if it did, call method provided in attribute value
        binding.value();
      }
    };
    document.addEventListener('click', el.clickOutsideEvent);
  },
  unmounted: (el: HTMLElementWithClickOutside) => {
    document.removeEventListener('click', el.clickOutsideEvent);
  },
};
