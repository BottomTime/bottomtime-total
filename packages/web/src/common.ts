import { AccountTier, LogBookSharing, ProfileDTO } from '@bottomtime/api';

import { ComputedRef } from 'vue';

export type Breadcrumb = {
  label: string | ComputedRef<string>;
  to?: string | ComputedRef<string>;
  active?: boolean;
};

export type Coordinates = {
  left: number;
  top: number;
  width: number;
  height: number;
};

export const DateTimeFormat = 'yyyy-MMM-dd hh:mm:ss a';

export const DefaultProfile: ProfileDTO = {
  accountTier: AccountTier.Basic,
  logBookSharing: LogBookSharing.Private,
  memberSince: 0,
  username: '',
  userId: '',
};

export type Selectable<T> = T & { selected?: boolean };

export type SelectOption = {
  disabled?: boolean;
  label?: string;
  value: string;
};

export type TabInfo = {
  key: string;
  label: string;
  disabled?: boolean;
  visible?: boolean;
};

export enum ToastType {
  Error = 'error',
  Info = 'info',
  Success = 'success',
  Warning = 'warning',
}

export type Toast = {
  id: string;
  message: string;
  type: ToastType;
};

export const FortyEightHoursInSeconds = 48 * 60 * 60 * 1000;
