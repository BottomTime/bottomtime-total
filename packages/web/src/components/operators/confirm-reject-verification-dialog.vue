<template>
  <DialogBase
    size="md"
    title="Reject Verification?"
    :disabled="isSaving"
    :visible="visible"
    @close="onCancel"
  >
    <template #default>
      <div class="flex gap-3">
        <p class="text-4xl">
          <i class="fa-regular fa-circle-question"></i>
        </p>

        <div class="space-y-3">
          <p>
            Do you want to reject the verification request for this dive shop?
            If so, it is advised that you provide a detailed reason for the
            rejection as well as steps for the operator to remedy the situation.
          </p>

          <FormField label="Reason" control-id="reject-reason">
            <FormTextArea
              v-model.trim="reason"
              control-id="reject-reason"
              test-id="reject-reason"
              autofocus
              :maxlength="1000"
            />
          </FormField>
        </div>
      </div>
    </template>
    <template #buttons>
      <FormButton
        test-id="btn-confirm-reject"
        type="danger"
        :is-loading="isSaving"
        submit
        @click="onConfirm"
      >
        Reject
      </FormButton>
      <FormButton test-id="btn-cancel-reject" @click="onCancel">
        Cancel
      </FormButton>
    </template>
  </DialogBase>
</template>

<script setup lang="ts">
import { onMounted, ref } from 'vue';

import FormButton from '../common/form-button.vue';
import FormField from '../common/form-field.vue';
import FormTextArea from '../common/form-text-area.vue';
import DialogBase from '../dialog/dialog-base.vue';

interface ConfirmRejectVerificationDialogProps {
  isSaving?: boolean;
  visible?: boolean;
}

withDefaults(defineProps<ConfirmRejectVerificationDialogProps>(), {
  isSaving: false,
  visible: false,
});
const emit = defineEmits<{
  (e: 'confirm', message?: string): void;
  (e: 'cancel'): void;
}>();

const reason = ref('');

function onConfirm() {
  emit('confirm', reason.value || undefined);
}

function onCancel() {
  reason.value = '';
  emit('cancel');
}

onMounted(() => {
  reason.value = '';
});
</script>
