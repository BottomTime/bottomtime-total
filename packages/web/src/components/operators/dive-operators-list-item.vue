<template>
  <li
    class="flex items-center gap-3 even:bg-blue-300/40 even:dark:bg-blue-900/40 rounded-md p-4"
  >
    <div class="w-[64px] h-[64px]">
      <img class="rounded-lg" :src="operator.logo" alt="" />
    </div>

    <article class="space-y-1">
      <div class="flex gap-2 align-top">
        <FormButton
          type="link"
          size="2xl"
          :test-id="`select-${operator.slug || operator.id}`"
          @click="$emit('select', operator)"
        >
          <span class="capitalize">{{ operator.name }}</span>
        </FormButton>

        <div class="relative group">
          <p v-if="operator.verified" class="text-sm text-success space-x-1">
            <span>
              <i class="fa-solid fa-circle-check fa-sm"></i>
            </span>
            <span
              class="absolute invisible group-hover:visible transition-opacity left-[105%] top-0 bg-success text-grey-950 px-2 rounded-md"
            >
              Verified!
            </span>
          </p>
        </div>
      </div>

      <p v-if="operator.description">
        {{ operator.description }}
      </p>

      <address class="grid grid-cols-2 px-2 gap-3 text-sm">
        <div class="space-y-1.5">
          <div v-if="operator.address || operator.gps" class="flex gap-2">
            <span>
              <i class="fa-solid fa-map-marker-alt"></i>
            </span>
            <label class="sr-only">Address</label>
            <p class="flex flex-col">
              <span v-if="operator.address">{{ operator.address }}</span>
              <span v-if="operator.gps">
                {{ `${operator.gps.lat}, ${operator.gps.lon}` }}
              </span>
            </p>
          </div>

          <div v-if="operator.phone" class="flex gap-2">
            <span>
              <i class="fa-solid fa-phone"></i>
            </span>
            <label class="sr-only">Phone</label>
            <NavLink :to="`tel:${operator.phone}`">
              {{ operator.phone }}
            </NavLink>
          </div>

          <div v-if="operator.email" class="flex gap-2">
            <span>
              <i class="fa-solid fa-envelope"></i>
            </span>
            <label class="sr-only">Email</label>
            <NavLink :to="`mailto:${operator.email}`">
              {{ operator.email }}
            </NavLink>
          </div>

          <div v-if="operator.website" class="flex gap-2">
            <span>
              <i class="fa-solid fa-globe"></i>
            </span>
            <label class="sr-only">Website</label>
            <NavLink :to="operator.website">
              {{ operator.website }}
            </NavLink>
          </div>
        </div>

        <div class="space-y-1.5">
          <div v-if="operator.socials?.facebook" class="flex gap-2">
            <span>
              <i class="fa-brands fa-facebook"></i>
            </span>
            <label class="sr-only">Facebook</label>
            <NavLink
              :to="`https://facebook.com/${operator.socials?.facebook}/`"
              new-tab
            >
              {{ operator.socials?.facebook }}
            </NavLink>
          </div>

          <div v-if="operator.socials?.instagram" class="flex gap-2">
            <span>
              <i class="fa-brands fa-instagram"></i>
            </span>
            <label class="sr-only">Instagram</label>
            <NavLink
              :to="`https://instagram.com/${operator.socials?.instagram}/`"
              new-tab
            >
              {{ operator.socials?.instagram }}
            </NavLink>
          </div>

          <div v-if="operator.socials?.tiktok" class="flex gap-2">
            <span>
              <i class="fa-brands fa-tiktok"></i>
            </span>
            <label class="sr-only">TikTok</label>
            <NavLink
              :to="`https://tiktok.com/@${operator.socials?.tiktok}/`"
              new-tab
            >
              {{ operator.socials?.tiktok }}
            </NavLink>
          </div>

          <div v-if="operator.socials?.twitter" class="flex gap-2">
            <span>
              <i class="fa-brands fa-x-twitter"></i>
            </span>
            <label class="sr-only">X / Twitter</label>
            <NavLink
              :to="`https://x.com/${operator.socials?.twitter}/`"
              new-tab
            >
              {{ operator.socials?.twitter }}
            </NavLink>
          </div>

          <div v-if="operator.socials?.youtube" class="flex gap-2">
            <span>
              <i class="fa-brands fa-youtube"></i>
            </span>
            <label class="sr-only">Youtube</label>
            <NavLink
              :to="`https://youtube.com/@${operator.socials?.youtube}/`"
              new-tab
            >
              {{ operator.socials?.youtube }}
            </NavLink>
          </div>
        </div>
      </address>
    </article>

    <div class="flex min-w-32">
      <FormButton v-if="false" rounded="left" size="sm">
        <p>
          <span class="sr-only">Edit {{ operator.name }}</span>
          <span>
            <i class="fa-solid fa-edit fa-sm"></i>
          </span>
        </p>
      </FormButton>
      <FormButton v-if="false" rounded="right" size="sm" type="danger">
        <p>
          <span class="sr-only">Delete {{ operator.name }}</span>
          <span>
            <i class="fa-solid fa-trash fa-sm"></i>
          </span>
        </p>
      </FormButton>
    </div>
  </li>
</template>

<script lang="ts" setup>
import { DiveOperatorDTO } from '@bottomtime/api';

import FormButton from '../common/form-button.vue';
import NavLink from '../common/nav-link.vue';

interface DiveOperatorsListItemProps {
  operator: DiveOperatorDTO;
}

defineProps<DiveOperatorsListItemProps>();
defineEmits<{
  (e: 'select', operator: DiveOperatorDTO): void;
}>();
</script>
