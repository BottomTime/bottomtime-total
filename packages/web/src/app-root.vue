<template>
  <SnackBar />
  <div
    id="modal"
    class="font-content text-grey-900 bg-blue-200 dark:text-grey-100 dark:bg-blue-950"
  ></div>
  <div id="drawer"></div>

  <div
    class="md:container mx-auto font-content text-grey-900 bg-blue-200 dark:text-grey-100 dark:bg-blue-950 rounded-b-xl shadow-md shadow-white dark:shadow-grey-700 opacity-95"
  >
    <!-- Placeholder element. Modal dialogs will be "teleported" here to get the z-indexing right. -->

    <section class="mt-16 p-4">
      <RouterView></RouterView>
    </section>
    <PageFooter />
  </div>
  <NavBar />
</template>

<script setup lang="ts">
import { onBeforeMount } from 'vue';

import { ToastType } from './common';
import NavBar from './components/core/nav-bar.vue';
import PageFooter from './components/core/page-footer.vue';
import SnackBar from './components/core/snack-bar.vue';
import { Config } from './config';
import { useErrors, useToasts } from './store';

const toasts = useToasts();
const errors = useErrors();

onBeforeMount(() => {
  if (!Config.isSSR) {
    if (errors.renderError) {
      if (!Config.isProduction) {
        /* eslint-disable-next-line no-console */
        console.error(errors.renderError);
      }
      toasts.toast({
        id: 'server-prerender-error',
        message:
          'A server error has occurred and the page may not have loaded correctly. We apologize for the inconvenience. Please try again later as we investigate the issue.',
        type: ToastType.Error,
      });
    }
  }
});
</script>
