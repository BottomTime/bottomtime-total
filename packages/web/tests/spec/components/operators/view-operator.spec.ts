import { ApiClient, VerificationStatus } from '@bottomtime/api';

import { ComponentMountingOptions, mount } from '@vue/test-utils';

import { Pinia, createPinia } from 'pinia';
import { ApiClientKey } from 'src/api-client';
import ViewOperator from 'src/components/operators/view-operator.vue';

import { FullOperator, PartialOperator } from '../../../fixtures/operators';

describe('ViewOperator component', () => {
  let client: ApiClient;

  let pinia: Pinia;
  let opts: ComponentMountingOptions<typeof ViewOperator>;

  beforeAll(() => {
    client = new ApiClient();
  });

  beforeEach(() => {
    pinia = createPinia();
    opts = {
      props: {
        operator: FullOperator,
      },
      global: {
        plugins: [pinia],
        provide: {
          [ApiClientKey as symbol]: client,
        },
      },
    };
  });

  it('will render with partial properties', async () => {
    const wrapper = mount(ViewOperator, opts);
    await wrapper.setProps({ operator: PartialOperator });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('will render with all properties', () => {
    const wrapper = mount(ViewOperator, opts);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('will show verified badge for verified operators', async () => {
    const wrapper = mount(ViewOperator, opts);
    await wrapper.setProps({
      operator: {
        ...FullOperator,
        verificationStatus: VerificationStatus.Verified,
      },
    });
    expect(wrapper.find('[data-testid="operator-verified"]').isVisible()).toBe(
      true,
    );
  });
});
