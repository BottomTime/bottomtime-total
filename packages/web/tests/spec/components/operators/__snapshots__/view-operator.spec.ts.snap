// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewOperator component will render with all properties 1`] = `
"<!--v-if-->
<!--v-if-->
<div class="flex gap-6" data-testid="view-operator-section">
  <div class="m-2 w-[128px] h-[128px]">
    <p><img src="https://loremflickr.com/128/128?lock=1873198725464064/128x128" class="rounded-md" width="128px" height="128px"></p>
  </div>
  <article class="space-y-3">
    <!--v-if-->
    <div class="line-clamp-4 hover:line-clamp-none">
      <!-- eslint-disable-next-line vue/no-v-html -->
      <div class="overflow-x-auto space-y-3">
        <p>Aufero accendo amissio suffoco adulescens cruentus despecto vulgaris. Valens delego coma eius universe. Delicate suffragium auditor debilito.
          Dolorum appositus vesper cinis vicissitudo vel cresco cibo brevis. Contabesco ulterius odit recusandae degero. Culpo animus adstringo dicta terebro eaque demo deporto defaeco.</p>
      </div>
      <!--v-if-->
    </div>
    <div class="grid grid-cols-1 lg:grid-cols-2 gap-4">
      <div class="space-y-2 px-3">
        <h3 class="text-lg font-bold font-title">Contact Info</h3>
        <p class="space-x-2"><span><i class="fa-solid fa-map-marker-alt fa-xs fa-fw"></i></span><span class="sr-only">Address</span><span>6713 Ritchie Key Apt. 697, Suffolk, LA, 43097</span></p>
        <p class="space-x-2"><span><i class="fa-solid fa-phone fa-xs fa-fw"></i></span><span class="sr-only">Phone Number</span><a href="tel:+19368100307;ext=63779">+1 936 810 0307 ext. 63779</a></p>
        <p class="space-x-2"><span><i class="fa-solid fa-envelope fa-xs fa-fw"></i></span><span class="sr-only">Email</span><a href="mailto:Elizabeth_Ernser@hotmail.com">Elizabeth_Ernser@hotmail.com</a></p>
        <p class="space-x-2"><span><i class="fa-solid fa-globe fa-xs fa-fw"></i></span><span class="sr-only">Website</span><a href="https://euphoric-controversy.com" target="_blank">https://euphoric-controversy.com</a></p>
      </div>
      <div class="space-y-2 px-3">
        <h3 class="text-lg font-bold font-title">Socials</h3>
        <p class="space-x-2"><span><i class="fa-brands fa-facebook fa-xs fa-fw"></i></span><span class="sr-only">Facebook</span><a href="https://www.facebook.com/Nestor34" target="_blank">Nestor34</a></p>
        <p class="space-x-2"><span><i class="fa-brands fa-instagram fa-xs fa-fw"></i></span><span class="sr-only">Instagram</span><a href="https://www.instagram.com/Warren7/" target="_blank">Warren7</a></p>
        <p class="space-x-2"><span><i class="fa-brands fa-tiktok fa-xs fa-fw"></i></span><span class="sr-only">TikTok</span><a href="https://www.tiktok.com/@Rickey.Steuber" target="_blank">Rickey.Steuber</a></p>
        <p class="space-x-2"><span><i class="fa-brands fa-x-twitter fa-xs fa-fw"></i></span><span class="sr-only">X</span><a href="https://x.com/Pierre_Rogahn" target="_blank">Pierre_Rogahn</a></p>
        <p class="space-x-2"><span><i class="fa-brands fa-youtube fa-xs fa-fw"></i></span><span class="sr-only">YouTube</span><a href="https://www.youtube.com/@squeaky_osmosis" target="_blank">squeaky_osmosis</a></p>
      </div>
    </div>
  </article>
</div>
<div class="space-y-3">
  <h1 class="text-2xl font-bold font-title">Reviews</h1>
  <div class="bg-blue-300 dark:bg-blue-900 p-3 rounded-md shadow-md shadow-grey-800 flex flex-col gap-1 md:flex-row justify-between items-center md:items-baseline sticky top-16 z-[40]">
    <p><span>Showing </span><span class="font-bold">0</span><span> of </span><span class="font-bold">0</span><span> reviews</span></p>
    <div class="flex flex-col gap-1 items-center md:items-end">
      <div class="flex flex-wrap gap-2 items-baseline">
        <p class="font-bold">Rating:</p>
        <div readonly="">0</div>
      </div>
      <div class="flex flex-col lg:flex-row items-center md:items-end lg:items-baseline gap-3">
        <div><select id="reviews-sort-order" class="bg-grey-200 border-2 border-grey-600 text-grey-950 rounded-lg block py-1 px-2 h-8 dark:bg-grey-300 dark:placeholder-grey-400 disabled:bg-grey-400 disabled:text-grey-700 disabled:dark:bg-grey-500 min-w-36" data-testid="reviews-sort-order" name="reviews-sort-order">
            <option value="createdAt-desc">Age (most recent to oldest)</option>
            <option value="createdAt-asc">Age (most recent to oldest)</option>
            <option value="rating-desc">Rating (highest to lowest)</option>
            <option value="rating-asc">Rating (lowest to highest)</option>
          </select></div>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class="text-center my-8 text-lg">
    <p class="space-x-3 italic"><span><i class="fas fa-spinner fa-spin"></i></span><span>Fetching customer reviews...</span></p>
  </div>
</div>"
`;

exports[`ViewOperator component will render with partial properties 1`] = `
"<!--v-if-->
<!--v-if-->
<div class="flex gap-6" data-testid="view-operator-section">
  <div class="m-2 w-[128px] h-[128px]">
    <p><i class="fa-regular fa-image fa-5x"></i></p>
  </div>
  <article class="space-y-3">
    <!--v-if-->
    <div class="line-clamp-4 hover:line-clamp-none">
      <!-- eslint-disable-next-line vue/no-v-html -->
      <div class="overflow-x-auto space-y-3">
        <p>Aufero accendo amissio suffoco adulescens cruentus despecto vulgaris. Valens delego coma eius universe. Delicate suffragium auditor debilito.
          Dolorum appositus vesper cinis vicissitudo vel cresco cibo brevis. Contabesco ulterius odit recusandae degero. Culpo animus adstringo dicta terebro eaque demo deporto defaeco.</p>
      </div>
      <!--v-if-->
    </div>
    <div class="grid grid-cols-1 lg:grid-cols-2 gap-4">
      <div class="space-y-2 px-3">
        <h3 class="text-lg font-bold font-title">Contact Info</h3>
        <p class="space-x-2"><span><i class="fa-solid fa-map-marker-alt fa-xs fa-fw"></i></span><span class="sr-only">Address</span><span>6713 Ritchie Key Apt. 697, Suffolk, LA, 43097</span></p>
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </article>
</div>
<div class="space-y-3">
  <h1 class="text-2xl font-bold font-title">Reviews</h1>
  <div class="bg-blue-300 dark:bg-blue-900 p-3 rounded-md shadow-md shadow-grey-800 flex flex-col gap-1 md:flex-row justify-between items-center md:items-baseline sticky top-16 z-[40]">
    <p><span>Showing </span><span class="font-bold">0</span><span> of </span><span class="font-bold">0</span><span> reviews</span></p>
    <div class="flex flex-col gap-1 items-center md:items-end">
      <div class="flex flex-wrap gap-2 items-baseline">
        <p class="font-bold">Rating:</p>
        <div readonly="">0</div>
      </div>
      <div class="flex flex-col lg:flex-row items-center md:items-end lg:items-baseline gap-3">
        <div><select id="reviews-sort-order" class="bg-grey-200 border-2 border-grey-600 text-grey-950 rounded-lg block py-1 px-2 h-8 dark:bg-grey-300 dark:placeholder-grey-400 disabled:bg-grey-400 disabled:text-grey-700 disabled:dark:bg-grey-500 min-w-36" data-testid="reviews-sort-order" name="reviews-sort-order">
            <option value="createdAt-desc">Age (most recent to oldest)</option>
            <option value="createdAt-asc">Age (most recent to oldest)</option>
            <option value="rating-desc">Rating (highest to lowest)</option>
            <option value="rating-asc">Rating (lowest to highest)</option>
          </select></div>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class="text-center my-8 text-lg">
    <p class="space-x-3 italic"><span><i class="fas fa-spinner fa-spin"></i></span><span>Fetching customer reviews...</span></p>
  </div>
</div>"
`;
